crates/runtime/src/io_runtime_extension.rs

```diff
@@ -0,0 +1,72 @@
+use cairo_felt::Felt252;
+use cairo_lang_runner::short_string::as_cairo_short_string;
+use cairo_vm::vm::{errors::hint_errors::HintError, vm_core::VirtualMachine};
+use cheatnet::{
+    cheatcodes::EnhancedHintError,
+    execution::cheatable_syscall_handler::{CheatableSyscallHandler, SyscallSelector},
+};
+
+use crate::{
+    CheatcodeHandlingResult, ExtendedRuntime, ExtensionLogic, RuntimeExtension,
+    SyscallHandlingResult,
+};
+
+pub struct IORuntimeState {}
+
+pub type IORuntime<'a> =
+    ExtendedRuntime<RuntimeExtension<IORuntimeState, CheatableSyscallHandler<'a>>>;
+
+impl<'a> ExtensionLogic for RuntimeExtension<IORuntimeState, CheatableSyscallHandler<'a>> {
+    type Runtime = CheatableSyscallHandler<'a>;
+
+    fn get_extended_runtime_mut(&mut self) -> &mut CheatableSyscallHandler<'a> {
+        &mut self.extended_runtime
+    }
+
+    fn get_extended_runtime(&self) -> &CheatableSyscallHandler<'a> {
+        &self.extended_runtime
+    }
+
+    fn handle_cheatcode(
+        &mut self,
+        selector: &str,
+        inputs: Vec<Felt252>,
+    ) -> Result<CheatcodeHandlingResult, EnhancedHintError> {
+        match selector {
+            "print" => {
+                print(inputs);
+                Ok(CheatcodeHandlingResult::Handled(vec![]))
+            }
+            _ => Err(EnhancedHintError::from(HintError::CustomHint(
+                "Only `print` cheatcode is available in contracts.".into(),
+            ))),
+        }
+    }
+
+    fn override_system_call(
```

Comment:
Isn't that doable with a default impl for a trait?

Reply:
Not sure if you can provide default impl for just one function that the trait requires :c

Reply:
DONE